src/snowflake/cli/plugins/object/stage/diff.py

```diff
@@ -154,7 +162,47 @@ def build_md5_map(list_stage_cursor: SnowflakeCursor) -> Dict[str, str]:
     }
 
 
-def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
+def _get_relative_paths_to_sync(
+    relative_files_to_sync: List[Path], local_path: Path, remote_paths: Set[str]
+) -> List[str]:
+    paths = []
+    for file in relative_files_to_sync:
+        path = Path(os.path.join(local_path, file))
```

Comment:
nit: mixing os.path and pathlib this way is weird. I'd create a Path object for the root dir outside of the loop, and just join the file to it for each iteration. Or just pass both args directly to the ctor, it will join them anyway.